### ISSUE TITLE
discarded_notification_center_observer false positive when adding to an array

### ISSUE THREAD

**Issue by mflider** [realm/SwiftLint#1739]
This produces a false positive:

```
var observers = [Any]()
observers.append(NotificationCenter.default.addObserver(/*blah blah*/))
```

It looks like SwiftLint isn't checking for the case where a notification observer object is being added directly to an array, instead of returned or assigned to a variable.

**Comment by marcelofabri** [comment]
Duplicate of #1398.
